pyacq/core/tests/test_tools.py

```diff
@@ -0,0 +1,70 @@
+from pyacq.core.stream  import OutputStream, InputStream
+from pyacq.core.tools  import ThreadPollInput
+from pyqtgraph.Qt import QtCore, QtGui
+import pyqtgraph as pg
+
+import numpy as np
+import weakref
+import time
+
+
+def test_ThreadPollInput():
+    app = pg.mkQApp()
+    
+    nb_channel = 16
+    chunksize = 1024
+    stream_spec = dict(protocol = 'tcp', interface = '127.0.0.1', port='*', 
+                       transfermode = 'plaindata', streamtype = 'analogsignal',
+                       dtype = 'float32', shape = (-1, nb_channel), compression ='',
+                       scale = None, offset = None, units = '', timeaxis = 0)
+    
+    class ThreadSender(QtCore.QThread):
+        def __init__(self, output_stream, parent = None):
+            QtCore.QThread.__init__(self, parent)
+            self.output_stream = weakref.ref(output_stream)
```

Comment:
No it is not possible in the actual design. See Node docstring.
Node.outputs[''].configure and Node.intputs[''].connect are done in between Node.configure() and Node.initialize().
So InputStream and OutPutStream belong to the Node. All node do not necessary have a thread.
Node.outputs[''].configure imply a socket.bind and Node.intputs[''].connect imply a socket.connect.
Threads are created at Node.initialize steps if necessary.

node.doc:
Usage of a Node, the **order** is important:
        1 - Instanciate : directly or remotly Node with NodeGroup.creat_node
        2 - Node.configure(...)
        3 - Connect Input (if have input) : Node.inputs['input_name'].connect(other_node.outpouts['output_name'])
        4 - Configure outputs : Node.outputs['output_name'].configure(...)
        5 - initialize : Node.initialize(), this also check inputs and outputs specifications.
        6 - Start/Stop many time : Node.start()/Node.stop()
        7 - Node.close() : done by NodeGroup if the Node is remoted